src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -186,10 +187,41 @@ public void start(final Map<String, String> props) {
         Optional<String> jsonDocument = Optional.empty();
         if (publishFullDocumentOnly) {
           if (changeStreamDocument.containsKey("fullDocument")) {
-            jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+            if (sourceConfig.getJsonType().equals("canonical")) {
+              jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+            } else if (sourceConfig.getJsonType().equals("relaxed")) {
+              jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson(JsonWriterSettings.builder()
+                      .outputMode(JsonMode.RELAXED)
+                      .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                      .build()));
+            } else {
+              try {
+              } catch (Exception e) {
+                LOGGER.info(
+                        "This Json format is not supported or Unknown please choose : \"relaxed\", \"canonical\": {}",
+                        e.getMessage());
+              }
+            }
           }
         } else {
-          jsonDocument = Optional.of(changeStreamDocument.toJson());
+          if (sourceConfig.getJsonType().equals("canonical")) {
+            jsonDocument = Optional.of(changeStreamDocument.toJson());
+          } else if (sourceConfig.getJsonType().equals("relaxed")) {
+            jsonDocument = Optional.of(changeStreamDocument.toJson(JsonWriterSettings.builder()
+                    .outputMode(JsonMode.RELAXED)
+                    .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                    .build()));
+          } else {
+
+            try {
+            } catch (Exception e) {
+              LOGGER.info(
+                      "This Json format is not supported or Unknown please choose : \"relaxed\", \"canonical\": {}",
```

Comment:
-> JSON

Reply:
There is no need of need now thanks to the `enum`